demisto_sdk/commands/generate_yml_from_python/tests/generate_yml_from_python_test.py

```diff
@@ -0,0 +1,1300 @@
+        """
+        Given
+        - Decorated command code with restored args usage.
+
+        When
+        - Running YMLGenerator generate.
+
+        Then
+        - Ensure restored args do not show up as command args in the generated YML dict.
+        """
+        integration_path = tmp_path / "integration_name.py"
+
+        def code_snippet():
+            metadata_collector = YMLMetadataCollector(integration_name="some_name")
+
+            @metadata_collector.command(command_name="funky-command", outputs_prefix="funk", execution=False,
+                                        outputs_list=[OutputArgument(name="out1", output_type=str,
+                                                                     description="desc1")],
+                                        inputs_list=[InputArgument(name="in1")], restore=True)
+            def funky_command(client, command_name, outputs_prefix, execution, args):
+                """Some funky command"""
+                print("func")
+
+        save_code_as_integration(code=code_snippet, full_path=integration_path)
+        yml_generator = YMLGenerator(filename=integration_path)
+        yml_generator.generate()
+        generated_dict = yml_generator.get_metadata_dict()
+
+        # Make sure something was generated.
+        assert "out1" in generated_dict["script"]["commands"][0]["outputs"][0]["contextPath"]
+
+        arguments = generated_dict["script"]["commands"][0]["arguments"]
+        args_names = [arguments[i]["name"] for i in range(len(arguments))]
+        for restored_arg in YMLMetadataCollector.RESTORED_ARGS:
+            assert restored_arg not in args_names
+
+    def test_restored_args(self, tmp_path):
+        """
+        Given
+        - Decorated command code with restored args usage.
+
+        When
+        - Running YMLGenerator generate.
+
+        Then
+        - Ensure restored args work properly when running the function without collecting metadata.
+        """
+        integration_path = tmp_path / "integration_name.py"
+
+        def code_snippet():
+            from demisto_sdk.commands.generate_yml_from_python.yml_metadata_collector import \
+                YMLMetadataCollector
+            metadata_collector = YMLMetadataCollector(integration_name="some_name")
+
+            @metadata_collector.command(command_name="funky-command", outputs_prefix="funk", execution=False,
+                                        restore=True)
+            def funky_command(client, command_name, outputs_prefix, execution):
+                """Some funky command"""
+                return client, command_name, outputs_prefix, execution
+
+        save_code_as_integration(code=code_snippet, full_path=integration_path)
+        yml_generator = YMLGenerator(filename=integration_path)
+        yml_generator.generate()
+        generated_dict = yml_generator.get_metadata_dict()
+        # Make sure something was generated
```

Comment:
can we improve this? like check some actual expected values?

Reply:
we do that in other unit tests , this is just a sanity check that the code ran properly